File: backend/src/main/java/com/school/management/entity/Semester.java
package com.school.management.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 学期实体类
 *
 * @author system
 * @since 2.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("semesters")
public class Semester extends BaseEntity {

    /**
     * 学期名称
     */
    private String semesterName;

    /**
     * 学期编码(如:2024-2025-2)
     */
    private String semesterCode;

    /**
     * 学期开始日期
     */
    private LocalDate startDate;

    /**
     * 学期结束日期
     */
    private LocalDate endDate;

    /**
     * 开始年份(如:2024)
     */
    @TableField(exist = false)
    private Integer startYear;

    /**
     * 学期类型:1-第一学期 2-第二学期
     */
    @TableField(exist = false)
    private Integer semesterType;

    /**
     * 是否当前学期:1-是 0-否
     */
    private Integer isCurrent;

    /**
     * 状态:1-正常 0-已结束
     */
    private Integer status;
}

File: frontend/src/types/v2/semester.ts
/**
 * 学期管理类型定义 - DDD架构适配
 */

/**
 * 学期类型枚举
 */
export type SemesterType = '第一学期' | '第二学期'

/**
 * 学期状态枚举
 */
export type SemesterStatus = '正常' | '已结束'

/**
 * 学期领域实体
 */
export interface Semester {
  id: number | string
  semesterName: string
  semesterCode: string
  startDate: string
  endDate: string
  startYear?: number
  semesterType?: SemesterType
  isCurrent?: boolean
  status?: SemesterStatus
  durationDays?: number
  isOngoing?: boolean
  createdAt?: string
  updatedAt?: string
}

/**
 * 创建学期请求
 */
export interface CreateSemesterRequest {
  semesterName: string
  startDate: string
  endDate: string
  startYear: number
  semesterType: number
}

/**
 * 更新学期请求
 */
export interface UpdateSemesterRequest {
  semesterName?: string
  startDate?: string
  endDate?: string
}

/**
 * 学期查询参数
 */
export interface SemesterQueryParams {
  pageNum?: number
  pageSize?: number
  semesterName?: string
  status?: number
}
